node-tests/blueprints/adapter-test.js

```diff
@@ -1,133 +1,107 @@
-      throws: {
-        message: /Adapters cannot extend from themself/,
-        type: 'SilentError'
-      }
-    });
+    var args = ['adapter', 'foo', '--base-class=foo'];
+
+    return emberNew()
+      .then(() => expect(emberGenerate(args))
+        .to.be.rejectedWith(SilentError, /Adapters cannot extend from themself/));
   });
 
   it('adapter when is named "application"', function() {
-    return generateAndDestroy(['adapter', 'application'], {
-      files: [
-        {
-          file: 'app/adapters/application.js',
-          contains: [
-            'import JSONAPIAdapter from \'ember-data/adapters/json-api\';',
-            'export default JSONAPIAdapter.extend({'
-          ]
-        },
-        {
-          file: 'tests/unit/adapters/application-test.js',
-          contains: [
-            'moduleFor(\'adapter:application\''
-          ]
-        }
-      ]
-    });
+    var args = ['adapter', 'application'];
+
+    return emberNew()
+      .then(() => emberGenerateDestroy(args, _file => {
+        expect(_file('app/adapters/application.js'))
+          .to.contain('import JSONAPIAdapter from \'ember-data/adapters/json-api\';')
+          .to.contain('export default JSONAPIAdapter.extend({');
+
+        expect(_file('tests/unit/adapters/application-test.js'))
+          .to.contain('moduleFor(\'adapter:application\'');
+      }));
   });
 
   it('adapter-test', function() {
-    return generateAndDestroy(['adapter-test', 'foo'], {
-      files: [
-        {
-          file: 'tests/unit/adapters/foo-test.js',
-          contains: [
-            'moduleFor(\'adapter:foo\''
-          ]
-        }
-      ]
-    });
+    var args = ['adapter-test', 'foo'];
+
+    return emberNew()
+      .then(() => emberGenerateDestroy(args, _file => {
+        expect(_file('tests/unit/adapters/foo-test.js'))
+          .to.contain('moduleFor(\'adapter:foo\'');
+      }));
   });
 
   it('adapter-test for mocha', function() {
-    return generateAndDestroy(['adapter-test', 'foo'], {
-      packages: [
-        { name: 'ember-cli-qunit', delete: true },
-        { name: 'ember-cli-mocha', dev: true }
-      ],
-      files: [
-        {
-          file: 'tests/unit/adapters/foo-test.js',
-          contains: [
-            'import { describeModule, it } from \'ember-mocha\';',
-            'describeModule(\n  \'adapter:foo\',',
-            'expect(adapter).to.be.ok;'
-          ]
-        }
-      ]
-    });
+    var args = ['adapter-test', 'foo'];
+
+    return emberNew()
+      .then(() => modifyPackages([
+        {name: 'ember-cli-qunit', delete: true},
+        {name: 'ember-cli-mocha', dev: true}
+      ]))
+      .then(() => emberGenerateDestroy(args, _file => {
+        expect(_file('tests/unit/adapters/foo-test.js'))
+          .to.contain('import { describeModule, it } from \'ember-mocha\';')
+          .to.contain('describeModule(\n  \'adapter:foo\',')
+          .to.contain('expect(adapter)' +
+          '.to.be.ok;');
```

Comment:
Can you combine this and the line above into 1 line, so it's consistent with the test for the transform test?